### DISCUSSION TITLE
dandi metadata: How should I link dandiset and asset meta-data in RDF

### DISCUSSION THREAD

**Discussion by surchs** [dandi/helpdesk#22] [Q&A] [answered]
I got the JSON-LD meta-data for a number of dandisets from the API. Let's say I want to find dandisets that are about `"Medial Temporal Lobe"` and for these count the number of participants with `sex=="Male"`. I think that the first is a dandiset level question, the second is an asset level question. 

I can get the assets for a dandiset from the API and convert them with reproschema to JSON-LD or something similar. For [dandiset 000004](https://gui.dandiarchive.org/#/dandiset/000004/draft) example IDs will look like this:

dandiset: `DANDI:000004/draft` where `DANDI` resolves to `http://dandiarchive.org/dandiset/`
asset: `dandiasset:38304fe9-5f37-4c0d-a741-9cf2bafab9ff` where `dandiasset` resolves to `http://dandiarchive.org/asset/`

Since neither contains the other's ID anywhere I think that when I feed them into my triple store they will be two unconnected graphs. What would be the intended way to connect them in a way that would allow me to answer the above question?

**Comment by satra** [comment, marked as answer]
we are adding a little collection file on publish (this just got merged), and we would be doing something similar in pushing draft metadata to a triplestore. 

here is the relevant issue: https://github.com/dandi/dandi-api/issues/399

you could temporarily, do the same for any assets you download. here is the relevant code in the api: https://github.com/dandi/dandi-api/blob/627eb4d617a9158c610cb59a0d4cb307833367ca/dandiapi/api/manifests.py#L93

effectively this information is there when you retrieve all the assets through the assets endpoint of a dandiset. also the metadata you receive from the api server for any asset or dandiset should already be jsonld, so you should not technically need to do any conversion.

**Reply by surchs** [reply]
Ah great, that is exactly what I'd need. I'll use your code example with `hasMember` to link them up for my local graph instance. Will this relationship (`hasMember`) eventually become part of the dandischema or is this going to be stored in `collections.jsonld`? In the latter case, what's the recommended way to get this file?

**Reply by satra** [reply]
don't know a good answer to that right this minute. probably will create an optional property for dandiset.
